lib/screens/calendar/day_tab.dart

```diff
@@ -0,0 +1,141 @@
+
+import 'package:admu_student_app/constants/app_colors.dart';
+import 'package:admu_student_app/constants/app_strings.dart';
+import 'package:admu_student_app/models/calendar_events.dart';
+import 'package:admu_student_app/models/event.dart';
+import 'package:admu_student_app/widgets/calendar/calendar.dart';
+import 'package:admu_student_app/widgets/calendar/event_card.dart';
+
+class DayTab extends StatefulWidget {
+  final DateTime date;
+  final Function(DateTime) onDateChange;
+
+  DayTab({this.date, this.onDateChange});
+
+  @override
+  _DayTabState createState() => _DayTabState();
+
+  
+}
+
+class _DayTabState extends State<DayTab> {
+  DateTime _date;
+  PageController _pageCtrl;
+
+  @override
+  void initState() {
+    super.initState();
+
+    if (widget.date != null)
+      _date = widget.date;
+    else
+      _date = DateTime.now();
+  }
+
+  String _getDateString() {
+    DateTime now = DateTime.now();
+
+    if (now.year == _date.year &&
+        now.month == _date.month &&
+        now.day == _date.day)
+      return 'Today';
+    else
+    // how to get the date strings 
+      return '${AppStrings.DAYS[_date.weekday - 1]}, ${AppStrings.MONTHS[_date.month - 1]} ${_date.day}';
+  }
+  
+ // Tests
+
+  dynamic _getDates() {
+  // Take the input year, month number, and pass it inside DateTime()
+  var now = DateTime.now();
+  
+  // Getting the total number of days of the month
+  var totalDays = daysInMonth(now);
+  
+  // Stroing all the dates till the last date
+  // since we have found the last date using generate
+  var listOfDates = new List<int>.generate(totalDays, (i) => i + 1);
+  return listOfDates;
+}
+
+dynamic _getNow() {
+  var now = DateTime.now();
+  return now.day - 1;
+}
+
+// this returns the last date of the month using DateTime
+int daysInMonth(DateTime date){
+  var firstDayThisMonth = new DateTime(date.year, date.month, date.day);
+  var firstDayNextMonth = new DateTime(firstDayThisMonth.year, firstDayThisMonth.month + 1, firstDayThisMonth.day);
+  return firstDayNextMonth.difference(firstDayThisMonth).inDays;
+}
+
+  @override
+  Widget build(BuildContext context) {
+    
+    // inside the Widget build, dispose it AND create a new controller - important for screen sizes
+    if (_pageCtrl != null) _pageCtrl.dispose();
+    _pageCtrl = PageController(
+        initialPage: _getNow(), // the date today 
+        viewportFraction: 150 / MediaQuery.of(context).size.width);
+    _pageCtrl.addListener(
+      () {
+        // check if _pageCtrl.page is an integer - means that the scrolling has locked into position
+        // you can set state here by setting the DateTime variable -- no need to do this one
+        print(_pageCtrl.page);
+      },
+    );
+
+    List<Event> _events =
+        Provider.of<CalendarEvents>(context).getEventsByDay(_date);
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        SizedBox(height: 27),
```

Comment:
i think the month + year (`June 2021`) is missing here

if i remember correctly, it should have a single `IconButton` to set the date (opens a new modal/dialog), but i forgot to create a separate method for it. the code is in `groups/input_date` or `select_date` but it needs to be edited a bit

what i would suggest is:
1. create two `IconButtons`, left and right of the text that simply changes the date
2. editing the date would be creating a new `DateTime` with `date.year, date.month +/- 1, 1` so that it auto lands on the first day

this goes against the design but it's up to you if you still want to copy and modify my code and do the singular icon button. hope it helps, just considering the time left (and that Sunday should actually be a rest day for everyone)

Reply:
ps., the code in `input_date` uses two different modals, one for web/android (material) and one for ios (cupertino). that's why my original code for the widget is messy